llarp/config/config.cpp

```diff
@@ -833,111 +833,162 @@ namespace llarp
         });
   }
 
-  LinksConfig::LinkInfo
-  LinksConfig::LinkInfoFromINIValues(std::string_view name, std::string_view value)
-  {
-    // we treat the INI k:v pair as:
-    // k: interface name, * indicating outbound
-    // v: a comma-separated list of values, an int indicating port (everything else ignored)
-    //    this is somewhat of a backwards- and forwards-compatibility thing
-
-    LinkInfo info;
-    info.port = 0;
-    info.addressFamily = AF_INET;
-
-    if (name == "address")
-    {
-      const IpAddress addr{value};
-      if (not addr.hasPort())
-        throw std::invalid_argument("no port provided in link address");
-      info.m_interface = addr.toHost();
-      info.port = *addr.getPort();
-    }
-    else
-    {
-      info.m_interface = std::string{name};
-
-      std::vector<std::string_view> splits = split(value, ",");
-      for (std::string_view str : splits)
-      {
-        int asNum = std::atoi(str.data());
-        if (asNum > 0)
-          info.port = asNum;
-
-        // otherwise, ignore ("future-proofing")
-      }
-    }
-
-    return info;
-  }
-
   void
   LinksConfig::defineConfigOptions(ConfigDefinition& conf, const ConfigGenParameters& params)
   {
-    constexpr Default DefaultOutboundLinkValue{"0"};
-
     conf.addSectionComments(
         "bind",
         {
-            "This section specifies network interface names and/or IPs as keys, and",
-            "ports as values to control the address(es) on which Lokinet listens for",
-            "incoming data.",
+            "Typically this section can be left blank, but can be used to specify which sockets to "
+            "bind on for inbound and outbound traffic.",
+            "",
+            "If no inbound bind addresses are configured then lokinet will search for a local ",
+            "network interface with a public IP address and use that IP with port 1090.",
+            "If no outbound bind addresses are configured then lokinet will use a wildcard "
+            "address.",
             "",
             "Examples:",
             "",
-            "    eth0=1090",
-            "    0.0.0.0=1090",
-            "    1.2.3.4=1090",
+            "    inbound=15.5.29.5:443",
+            "    inbound=10.0.2.2",
+            "    outbound=0.0.0.0:9000",
```

Comment:
Should add support for `inbound=:1234` that does the same go-find-an-ip behaviour, but overrides the default port to 1234 instead of 1090.

Reply:
should be in now